### PR TITLE
Remove inconsistant mutation error

### LINKED ISSUE #1180: Support silent mutations
Silent mutations are mutations which don't change the state i.e., two consecutive mutations with derived_state = "A".

Currently, support for these mutations is in a bit of grey area. We don't detect their existence at load time (see #1152) but we will throw an error when someone tries to decode the genotypes.

The original motivation for throwing this error was to help people find errors in their tree sequences - silent mutations seemed like something you would only do by mistake. However, it turns out that there's good mathematical reasons for wanting to have silent mutations from the perspective of simulating mutational processes (https://github.com/tskit-dev/msprime/issues/1440) and in particular trying to overlay multiple mutational processes is made hugely more complex if we **don't** have silent mutations. 

The changes we need to make should be minor:

- Remove the error raised during genotypes
- Update the test suite to make sure we have coverage of using silent mutations in various places (e.g., in the stats api)
- Update docs to clarify what silent mutations are and their support status.

See also: #1152, #325, https://github.com/tskit-dev/msprime/issues/1440

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #1180 
WIP, needs stats tests

### CODE REVIEW

File: c/tskit/core.h [hunk 1]
```diff
@@ -247,7 +247,6 @@ not found in the file.
 #define TSK_ERR_MUTATION_PARENT_EQUAL                               -501
 #define TSK_ERR_MUTATION_PARENT_AFTER_CHILD                         -502
 #define TSK_ERR_MUTATION_PARENT_INCONSISTENT                        -503
-#define TSK_ERR_INCONSISTENT_MUTATIONS                              -504
```
Comment (petrelharp) [inline c/tskit/core.h:250]: should probably decrement subsequent error codes?

Comment (jeromekelleher) [reply c/tskit/core.h:250]: I'm good either way - we definitely won't do that sort of thing after 1.0, but we're not there yet so :shrug:

File: c/tskit/genotypes.c [hunk 2]
```diff
@@ -333,7 +330,7 @@ tsk_vargen_update_genotypes_i8_sample_list(
             index = list_next[index];
         }
     }
-out:
+
```
Comment (jeromekelleher) [inline c/tskit/genotypes.c:333]: These functions can no long error, so we should make them void. Normally wouldn't bother, but this is perf sensitve.

**Review by jeromekelleher — COMMENTED** [review]
LGTM, minor comments

File: c/tskit/genotypes.c (hunk 2 above)
Comment (benjeffery) [reply c/tskit/genotypes.c:333]: The return value appears to be a count of the number of missing genotypes, which is used to set the `has_missing_data` flag.

File: c/tskit/core.h (hunk 1 above)
Comment (benjeffery) [reply c/tskit/core.h:250]: I've renumbered in https://github.com/tskit-dev/tskit/pull/1233/commits/614c001afe1d468b9c3698a2c4196e415a910202

File: c/tskit/genotypes.c (hunk 2 above)
Comment (jeromekelleher) [reply c/tskit/genotypes.c:333]: Ah - my bad.

**Comment by jeromekelleher** [comment]
Good, that's a sign it's working! Let me know if you have any questions/would like me to take a look at anything.

**Comment by benjeffery** [comment]
Ah, the issue is that a site with only silent mutations has a blank vcf `ALT` column. I guess it should be `.`.

**Comment by benjeffery** [comment]
Last failure is due to this change exposing #1225 to the test suite. Will fix that in seperate PR and merge before this one.

**Comment by petrelharp** [comment]
Oh dear, this means I should finish up https://github.com/tskit-dev/tskit/pull/1226

**Comment by jeromekelleher** [comment]
What's the status here @benjeffery? Should we skip some tests here in this PR so we can merge it, then rebase #1226 and fix there?

**Comment by benjeffery** [comment]
This only needs #1225, which I'm working on. Should be done in soon, so will just wait and rebase.

**Comment by benjeffery** [comment]
#1225 is getting more complicated than I realised so have added a skip here.

**Review by jeromekelleher — APPROVED** [review]
LGTM

**Pull request merged**